docs/shared-constraints.md

```diff
@@ -0,0 +1,69 @@
+# Shared Constraints
+
+Custom constraints in `protovalidate` afford a lot of power, but can become
+cumbersome and repetitive when the same kind of custom constraints are needed
+across multiple fields or messages. To this end, `protovalidate` provides a
+mechanism for creating reusable constraints that can be applied on multiple
+fields.
+
+Shared constraints require Protobuf extensions, which are not available in
+proto3. Either proto2 or at least Protobuf 2023 Edition must be used to define
+shared constraints. Shared constraints defined in proto2 or Protobuf 2023
+Edition or later can be imported and utilized in proto3 files.
+
+## Shared Field Constraints
+
+To define a shared field rule, extend one of the standard rules messages. For
+example, to define a new rule for `float` fields, extend
+`buf.validate.FloatRules`, as follows:
+
+```proto
+import "buf/validate/shared/constraints.proto";
```

Comment:
Note that these docs no longer match impl

Reply:
Documentation has been updated.